ControlFlowArraysAndStrings/ControlFlowArraysAndStrings/Program.cs

```diff
@@ -0,0 +1,343 @@
+                {
+                    newArray[i] = array[i];
+                }
+
+                if (i >= index)
+                {
+                    newArray[i] = array[i + 1];
+                }
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Attempts to log in a user and prints a message to the console if login 
+        /// was successful or unsuccessful. Allows 3 unsuccessful attempts before lock out.
+        /// </summary>
+        public static void Login()
+        {
+            int loginAttempts = 3;
+            string[] credentialType = { "user name", "password"};
+            bool[] isValid = { false, false };
+
+            for (int i = 0; i < credentialType.Length; i++)
+            {
+                while (!isValid[i] && loginAttempts != 0)
+                {
+                    string credential = GetLoginCredentialsFromUser(credentialType[i]);
+                    isValid[i] = ValidateCredential(credentialType[i], credential);
+
+                    if (!isValid[i])
+                    {
+                        loginAttempts--;
+                        Console.WriteLine($"Invalid {credentialType[i]}. {loginAttempts} tries left.");
+                    }
+
+                    if (loginAttempts == 0)
+                    {
+                        Console.WriteLine("You have been locked out. Please contact your system administrator.");
+                        break;
+                    }
+                }
+            }
+
+            if (isValid[0] && isValid[1])
+            {
+                Console.WriteLine("Logged in!");
+            }
+        }
+
+        /// <summary>
+        /// Gets the login credentials from the user.
+        /// </summary>
+        /// <param name="credentialType">The type of credential.</param>
+        /// <returns>The user's credential.</returns>
+        public static string GetLoginCredentialsFromUser(string credentialType)
+        {
+            Console.Write($"Please enter your {credentialType}: ");
+            return Console.ReadLine();
+        }
+
+        /// <summary>
+        /// Validates the specified user name or password against a hidden value.
+        /// </summary>
+        /// <param name="credentialType">The type of credential.</param>
+        /// <param name="credential">The credential to be validated.</param>
+        /// <returns>True if valid, or false if invalid.</returns>
+        public static bool ValidateCredential(string credentialType, string credential)
+        {
+            bool isValid = false;
+
+            if (credentialType.Equals("user name", StringComparison.OrdinalIgnoreCase))
+            {
+                string hiddenUsername = "CSharp";
+                isValid = credential.Equals(hiddenUsername,
+                                            StringComparison.OrdinalIgnoreCase);
+            }
+
+            if (credentialType.Equals("password", StringComparison.OrdinalIgnoreCase))
```

Comment:
This is not correct.  The requirements ask you to compare passwords case sensitively.